wire-ios-sync-engine/Tests/Source/Use cases/CreateConversationGuestLinkUseCaseTests.swift

```diff
@@ -0,0 +1,127 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModelSupport
+import XCTest
+
+@testable import WireSyncEngine
+
+final class CreateConversationGuestLinkUseCaseTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private let coreDataStackHelper = CoreDataStackHelper()
+    private var stack: CoreDataStack!
+    private let modelHelper = ModelHelper()
+    private var mockConversation: ZMConversation!
+    private var mockSelfUser: ZMUser!
+    private var sut: CreateConversationGuestLinkUseCaseProtocol!
+    private var setAllowGuestAndServicesUseCase: SetAllowGuestAndServicesUseCaseProtocol!
+
+    private var syncContext: NSManagedObjectContext {
+        return stack.syncContext
+    }
+
+    // MARK: - setUp
+
+    override func setUp() async throws {
+        try await super.setUp()
+        stack = try await coreDataStackHelper.createStack()
+        await syncContext.perform { [self] in
+            setAllowGuestAndServicesUseCase = SetAllowGuestAndServicesUseCase()
+            sut = CreateConversationGuestLinkUseCase(setGuestsAndServicesUseCase: setAllowGuestAndServicesUseCase)
+            mockSelfUser = modelHelper.createSelfUser(in: syncContext)
+            mockConversation = modelHelper.createGroupConversation(in: syncContext)
+            mockConversation.teamRemoteIdentifier = UUID()
+        }
+    }
+
+    // MARK: - tearDown
+
+    override func tearDown() async throws {
+        stack = nil
+        sut = nil
+        try coreDataStackHelper.cleanupDirectory()
+        try await super.tearDown()
+    }
+
+    // MARK: - Unit Tests
+
+    func testThatLinkGenerationSucceeds() async {
+
+        await syncContext.perform { [self] in
+            // GIVEN
+            let role = Role.insertNewObject(in: syncContext)
+            let action = Action.insertNewObject(in: syncContext)
+            role.name = "wire_admin"
+            action.name = "modify_conversation_access"
+            role.actions = [action]
+
+            mockConversation.addParticipantAndUpdateConversationState(user: mockSelfUser, role: role)
+
+            let mockHandler = MockActionHandler<CreateConversationGuestLinkAction>(result: .success("www.test.com"), context: syncContext.notificationContext)
+
+            let expectation = XCTestExpectation(description: "Guest link creation")
+
+            sut.invoke(conversation: mockConversation, password: nil) { result in
+                switch result {
+                case .success(let link):
+                    XCTAssertNotNil(link)
+                case .failure(let error):
+                    XCTFail("Test failed with error: \(error)")
+                }
+
+                expectation.fulfill()
+            }
+
+            wait(for: [expectation], timeout: 4.0)
```

Comment:
4 seconds is a bit long, usually we wait for 0.5 seconds.

Reply:
Done!